components/dfplayer_mini.rst

```diff
@@ -0,0 +1,397 @@
+
+Overview
+--------
+
+The module can be powered by the 3V output of an NodeMCU and a powered speaker connected to
+the modules ``DAC_R``, ``DAC_I`` and ``GND`` . You can connect only the ``tx_pin`` of
+the ``uart`` bus to the module's ``RX`` but if you need feedback of playback active you will
+also need to connect the ``rx_pin`` to the module's ``TX``.
+
+.. code-block:: yaml
+
+    # Example configuration entry
+    uart:
+      tx_pin: GPIO2
+      rx_pin: GPIO5
+      baud_rate: 9600
+
+    # Declare DFPlayer mini module
+    dfplayer:
+      on_finished_playback:
+        then:
+          logger.log: 'Somebody press play!'
+
+Configuration variables:
+------------------------
+
+- **uart_id** (*Optional*, :ref:`config-id`): Manually specify the ID of the UART hub.
+- **id** (*Optional*, :ref:`config-id`): Manually specify the ID used for code generation.
+- **on_finished_playback** (*Optional*, :ref:`Automation <automation>`): An action to be
+  performed when playback is finished.
+
+``dfplayer.is_playing`` Condition
+---------------------------------
+
+This Condition returns true while playback is active.
+
+.. code-block:: yaml
+
+    # In some trigger:
+    on_...:
+      if:
+        condition:
+          dfplayer.is_playing
+        then:
+          logger.log: 'Playback is active!'
+
+
+``dfplayer.play_next`` Action
+-----------------------------
+
+Starts playback of next track or skips to the next track.
+
+.. code-block:: yaml
+
+    on_...:
+      then:
+        - dfplayer.play_next:
+
+``dfplayer.play_previous`` Action
+---------------------------------
+
+Plays the previously played track.
+
+.. code-block:: yaml
+
+    on_...:
+      then:
+        - dfplayer.play_previous:
+
+
+``dfplayer.play`` Action
+------------------------
+
+Plays a track.
+
+.. code-block:: yaml
+
+    on_...:
+      then:
+        - dfplayer.play:
+          file: 23
+          loop: false
+        # Shorthand
+        - dfplayer.play: 23
+
+Configuration options:
+
+- **file** (*Optional*, int, :ref:`templatable <config-templatable>`): The global track
+  number (from all tracks in the device). If not specified plays the first track.
+- **loop** (*Optional*, bool, :ref:`templatable <config-templatable>`): Repeats playing
+  the same track. Defaults to ``False``.
+
+``dfplayer.play_folder`` Action
+-------------------------------
+
+Plays files inside numbered folders, folders must be numbered from 1 and with leading
+zeros. Like `01`, `02`, ... etc. Files inside the folders must be numbered with two
+leading zeros, like `001.mp3`, `002.mp3`, ... etc.
+Folder numbers can range from 1 to 99 and file name from 1 to 255 or folder number
+from 1 to 10 and file number from 1 to 1000.
+
+.. code-block:: bash
+
+    /01
+      /001.mp3
+      /002.mp3
+      ..
+    /02
+      /001.mp3
+      /002.mp3
+      /003.mp3
+      ..
+
+.. code-block:: yaml
+
+    on_...:
+      then:
+        - dfplayer.play_folder:
+          folder: 2
+          file: 1
```

Comment:
folder and file needs more spaces :rocket:

Reply:
ahh thanks